File: testing/CPP/local/src/sender.cpp
/*******************************************************************************
FILE:			sender.cpp

DATE:			February 4, 2025

DESCRIPTION:	This file sends messages to receiver.c, poller.c, selector.c,
				and senrec.c.

AUTHOR:			John Collins
*******************************************************************************/

#include <cstdlib>
#include <iostream>
#include <string>
#include <sim.h>

using namespace std;

int main(int argc, char **argv)
{
const int memLimit = 1024;	
int receiverId, limit, in[memLimit], out[memLimit];
string hname, rname;

if (argc != 4)
	{
	cout << "incorrect cmd line: sender senderName receiverName #";
	cout << "how many times?" << endl;
	exit(EXIT_FAILURE);
	}

SRY nee(argv[1]);

rname = argv[2];

if ((receiverId = nee.Locate(hname, rname, sizeof out, SIM_LOCAL)) == -1)
	{
	cout << "Can't locate receiver " << rname << endl;
	exit(EXIT_FAILURE);
	}

limit = atoi(argv[3]);

// compose the message
for (auto i : out)
	i = i;

for (int i= 0; i < limit; ++i)
	{
	if (nee.Send(receiverId, out, sizeof out, in, sizeof in) == -1)
		{
		cout << "Failed send" << endl;
		exit(EXIT_FAILURE);
		}
	usleep(10);
	}

return 0;
}

File: testing/python/Py/sender.py
"""
FILE:			sender.py

DATE:			February 4, 2025

DESCRIPTION:	This program constructs a message using short method, then sends the 					message to a receiver, then deconstructs the reply message using the 					short method and then prints the message contents to the screen. 						It does this 100 times.

                Python 3 compatible only.

AUTHOR:			John Collins
"""

# import required modules
import sys
import sim
import testMsg as msg

# set this program's name
sName = "SENDER_P"

# set the receiver program's name
rName = "SENREC_P"

# create an instance of a sry sender
nee = sim.SRY(sName, 1024);

# name locate the receiver program
receiverId = nee.locate("", rName, 1024, nee.SIM_LOCAL)
if receiverId == -1:
	print(sName, ": name locate error")
	sys.exit(-1)

for ctr in range(1, 101, 1):
	# build send message
	msg.constructMsg_S(nee, 1, nee.BIN)

	# send the message onward
	retVal = nee.send(receiverId)
	if retVal == -1:
		print(sName, ": send error")
		sys.exit(-1)

	# print replied message
	msg.deconstructMsg_S(nee, sName, nee.BIN)

	# print the loop number
	print("-------------" , " loop ", ctr, " -------------\n\n")
